ui/desktop/src/components/settings/providers/modal/ProviderConfiguationModal.tsx

```diff
@@ -53,6 +54,95 @@ export default function ProviderConfigurationModal() {
       setValidationErrors({});
       setShowDeleteConfirmation(false);
       setIsActiveProvider(false); // Reset active provider state
+
+      // If this is a custom provider, fetch the full provider JSON to pre-fill extra fields
+      if (currentProvider.name.startsWith('custom_')) {
+        (async () => {
+          try {
+            const secretKey = await window.electron.getSecretKey();
```

Comment:
is there a reason we can't use the generated client code here? we shouldn't need to call getSecret ever